scripts/Muon_Analysis_2.py

```diff
@@ -11,21 +11,82 @@
 
 import PyQt4.QtGui as QtGui
 import PyQt4.QtCore as QtCore
-
+from mantid.kernel import ConfigServiceImpl
+from Muon.GUI.Common.muon_data_context import MuonDataContext
 from Muon.GUI.Common.dummy_label.dummy_label_widget import DummyLabelWidget
 from Muon.GUI.MuonAnalysis.dock.dock_widget import DockWidget
 from Muon.GUI.Common.muon_context.muon_context import *  # MuonContext
 from save_python import getWidgetIfOpen
+from Muon.GUI.MuonAnalysis.load_widget.load_widget import LoadWidget
+import Muon.GUI.Common.message_box as message_box
+from Muon.GUI.Common.muon_load_data import MuonLoadData
+
 
 Name = "Muon_Analysis_2"
 
+
 muonGUI = None
+SUPPORTED_FACILITIES = ["ISIS", "SmuS"]
+
+
+def check_facility():
+    """
+    Get the currently set facility and check if it is in the list
+    of supported facilities, raising an AttributeError if not.
+    """
+    current_facility = ConfigServiceImpl.Instance().getFacility().name()
+    if current_facility not in SUPPORTED_FACILITIES:
+        raise AttributeError("Your facility {} is not supported by MuonAnalysis 2.0, so you"
+                             "will not be able to load any files. \n \n"
+                             "Supported facilities are :"
+                             + "\n - ".join(SUPPORTED_FACILITIES))
+
+
+class MuonAnalysis4Gui(QtGui.QMainWindow):
```

Comment:
4?

Reply:
I've tidied this function up by removing the multiple versions

Reply:
I thought meant version 4 so found it confusing as we are building version 2